mypy/semanal.py

```diff
@@ -427,12 +431,15 @@ def visit_func_def(self, defn: FuncDef) -> None:
                     # has external return type `Awaitable[T]`.
                     defn.type = defn.type.copy_modified(
                         ret_type = self.named_type_or_none('typing.Awaitable',
-                                                           [defn.type.ret_type]))
+                                                           [defn.type.ret_type]) or
+                        # We are running tests
+                        self.named_type('typing_full.Awaitable', [defn.type.ret_type]))
```

Comment:
This should be indented more (and maybe rearranged; I think it's too much for one statement).